### ISSUE TITLE
The api forward defined in .roadhogrc.mock.js dosn't take effects.

### ISSUE THREAD

**Issue by zheeeng** [sorrycc/roadhog#255]
#### Environment

- dva 0.7.8
- roadhog ^0.5.2
- Nodejs v7.5.0, Npm 4.1.2
- Operating environment MacOS 10.11.6

#### What did you do? Please provide steps to re-produce your problem.

Use the dva-cli create a new project.

Add mock rules:

```
export default {
  'GET /api/users': { users: [1,2] },    // pass

  '/api/users/1': { id: 1 },  // pass

  'POST /api/users/create': (req, res) => { res.end('OK'); },  // pass

  'GET /assets/*': 'https://assets.online/',  // doesn't take effect and break the entire mock rules defined. [serious!!]

  'GET /someDir/(.*)': 'https://g.alicdn.com/tb-page/taobao-home', // doesn't take effect and break the entire mock rules defined. [serious!!]
};

```

#### What do you expected?

The mock forward feature can forward local api to backend server.

#### What happen?

Once defined the forward rule, the mock feature be broken. BTW, in the project, we'd better change the script src in public/index.html from 'index.js' to '/index.js', other wise any url with slash such as 'aaa/bbb'  will lead the page can't find script file 'aaa/bbb/index.js'.

#### Re-producible online demo

Just use the dva-cli to create a new project and add forward rules to .roadhogrc.mock.js, it will reproduce.

**Comment by sorrycc** [comment]
yes, api forword has some problem with the order of express middleware
